app/src/lib/stores/app-store.ts

```diff
@@ -3346,17 +3346,29 @@ export class AppStore extends TypedBaseStore<IAppState> {
   public async _deleteBranch(
     repository: Repository,
     branch: Branch,
-    includeRemote?: boolean
+    includeUpstream?: boolean
   ): Promise<void> {
     return this.withAuthenticatingUser(repository, async (r, account) => {
       const gitStore = this.gitStoreCache.get(r)
 
       // If solely a remote branch, there is no need to checkout a branch.
       if (branch.type === BranchType.Remote) {
-        await gitStore.performFailableOperation(() => {
-          // Note: deleting a remote branch implementation not needed, yet.
-          return Promise.resolve()
-        })
+        if (branch.remoteName === null) {
+          // This is based on the branches ref. It should not be null for a
+          // remote branch
+          throw new Error(
+            `Could not determine remote name from: ${branch.ref}.`
+          )
+        }
+
+        await gitStore.performFailableOperation(() =>
+          deleteRemoteBranch(
+            r,
+            account,
+            branch.remoteName!,
+            branch.nameWithoutRemote
+          )
+        )
```

Comment:
I think that while the use in `!` is kind of acceptable in tests, we shouldn't use it in production code.
If I understand correctly, you need it because the `if` above is checking on a computed property that might change its value. If so, I've seen this pattern in other parts of the code to prevent the use of `!`:

```suggestion
        const remoteName = branch.remoteName
        if (remoteName === null) {
          // This is based on the branches ref. It should not be null for a
          // remote branch
          throw new Error(
            `Could not determine remote name from: ${branch.ref}.`
          )
        }

        await gitStore.performFailableOperation(() =>
          deleteRemoteBranch(
            r,
            account,
            remoteName,
            branch.nameWithoutRemote
          )
        )
```

Reply:
Not sure why the diff is so bad 😞 basically I extracted the `remoteName` to a constant, and thanks to that the compiler will help you out to avoid `!`.

Reply:
Ah - didn't realize using the ! was not ideal. Thanks! I updated per your suggestion.